effekt/shared/src/main/scala/effekt/generator/llvm/Transformer.scala

```diff
@@ -273,6 +283,16 @@ object Transformer {
         emit(FAdd(name, ConstantDouble(x), ConstantDouble(0)));
         transform(rest)
 
+      case machine.LiteralUTF8String(v@machine.Variable(bind, _), utf8, rest) =>
+        MC.staticText += s"$bind.lit" -> utf8
+        emit(RawLLVM(s"""
+%$bind.lit.decayed = bitcast [${utf8.length} x i8]* @$bind.lit to i8*
+%$bind = call %Pos @c_buffer_construct(i64 ${utf8.size}, i8* %$bind.lit.decayed)
+"""))
+        val transformed = transform(rest)
+        eraseValue(v) // TODO Is this the correct point in time to decrease the ref count by one?
```

Comment:
This would be too late, it should be:
```
eraseValues(List(v), freeVariables(rest));
transform(rest)
```

Reply:
Fixed: [aab95a2](https://github.com/effekt-lang/effekt/pull/142/commits/aab95a26e67b6dfb43f8966be19e3a4a469a48fb)